src/kubernetes.cc

```diff
@@ -961,6 +961,34 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidateConfiguration() const {
+  try {
+    (void) QueryMaster(
+        std::string(kKubernetesEndpointPath) + "/nodes?limit=1");
+
+    return true;
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+
+  const std::string node_name = CurrentNode();
+  if (node_name.empty()) {
+    return false;
+  }
+  
+  try {
+    (void) QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1");
```

Comment:
We should try to use `KubernetesPodLabelSelector` to see if it gets rejected (bad syntax, etc).

Reply:
I'm really not completely comfortable with this, as technically they could pass in garbage for a node name as well. I strongly feel the intention of this code should not be a unit test, but rather a confirmation that the endpoint is available under normal scenarios. What we have so far is better than nothing, and I would argue we're going too far down a path with too many assumptions by incorporating user-defined values as a part of this logic.

Reply:
We should validate all user-supplied configuration. That said, I'm ok with this being done in a later PR.